src/com/thunder_cut/server/SyncServer.java

```diff
@@ -65,84 +69,68 @@ public void run() {
             }
         }
     }
 
-    /**
-     * Check dataType and decide writeMode
-     * If type is command, send data to srcID
-     * or write to everyone
-     *
-     * @param src  client who send data
-     * @param type data type
-     * @param data pure data(No header)
-     */
-    private void identifyWriteMode(ClientInformation src, DataType type, byte[] data) {
-        if (type == DataType.CMD) {
-            send(src, type, data, src);
-        } else {
-            send(src, type, data);
-        }
-    }
-
-    /**
-     * Generate with srcID, data type, pure data and Send to a specific client.
-     *
-     * @param src  client who send data
-     * @param type data type
-     * @param data pure data(No header)
-     * @param dest
-     */
-    public void send(ClientInformation src, DataType type, byte[] data, ClientInformation dest) {
-        SendingData sendingData = new SendingData(clientGroup.indexOf(src), clientGroup.indexOf(dest), type, data);
-        try {
-            dest.getClient().write(sendingData.toByteBuffer());
-        } catch (IOException e) {
-            disconnected(dest);
-        }
-    }
-
-    /**
-     * Generate with srcID, data type, pure data and Send to everyone in clientGroup
-     *
-     * @param src  client who send data
-     * @param type data type
-     * @param data pure data(No header)
-     */
-    public void send(ClientInformation src, DataType type, byte[] data) {
-        for (Iterator<ClientInformation> iterator = clientGroup.iterator(); iterator.hasNext(); ) {
-            ClientInformation dest = iterator.next();
-            send(src, type, data, dest);
-        }
-    }
-
     @Override
-    public void received(ClientInformation client, DataType type, byte[] data) {
-        identifyWriteMode(client, type, data);
+    public void received(ReceivedData data) {
+        process.getProcessMap().get(DataType.valueOf(data.getDataType().type)).accept(data, clientMap);
     }
 
+    //TODO Resolve Bug (Read Git Issue), When disconnect change client
     /**
      * Remove disconnected client in clientGroup
      * After that change client's ID by ascending sort
      *
      * @param client disconnected client
      */
     @Override
-    public void disconnected(ClientInformation client) {
+    public void disconnected(ClientInfo client) {
         try {
             System.out.println(client.getClient().getRemoteAddress() + " is disconnected.");
             client.getClient().close();
             synchronized (clientGroup) {
                 clientGroup.remove(client);
             }
+            synchronized (clientMap) {
+                clientMap.remove(client);
+                for (ClientInfo information : clientMap.keySet()) {
+                    clientMap.get(information).remove(client);
+                }
+            }
         } catch (Exception e) {
             e.printStackTrace();
         }
     }
+
+    private void connectionClear(){
```

Comment:
Rename from _connectionClear()_ to _clearConnection()_.

- _valueGet()_ **X**
- _getValue()_ **O**

- _connectionClear()_ **X**
- _clearConnection()_ **O**

Please follow the **naming convention**.